src/services/completions.ts

```diff
@@ -1758,4 +1758,30 @@ namespace ts.Completions {
                 return node.parent;
         }
     }
+
+    /**
+     * Gets all properties on a type, but if that type is a union of several types,
+     * tries to only include those types which declare properties, not methods.
+     * This ensures that we don't try providing completions for all the methods on e.g. Array.
+     */
+    function getPropertiesForCompletion(type: Type, checker: TypeChecker): Symbol[] {
+        if (type.flags & TypeFlags.Union) {
+            const propertyOnlyTypes = mapDefined((type as UnionType).types, memberType => {
+                if (memberType.flags & TypeFlags.Primitive) {
+                    return undefined;
+                }
+                if (checker.getPropertiesOfType(memberType).some(p =>
+                    isMethodDeclaration(p.valueDeclaration) || isMethodSignature(p.valueDeclaration))) {
+                    return undefined;
+                }
+                return memberType;
+            });
+            // If there are no property-only types, just provide completions for every type as usual.
+            if (propertyOnlyTypes.length > 0) {
+                type = checker.getUnionType(propertyOnlyTypes);
+            }
+        }
+
+        return checker.getAllPossiblePropertiesOfType(type);

```

Comment:
can we instead have `getAllPossiblePropertiesOfTypes` that takes an array of types? i think this is the only reference to the internal API `getAllPossiblePropertiesOfType` anyways.